fix(rdb): add upgrade info (#4768)
scaleway/docs-content: pages/managed-databases-for-postgresql-and-mysql/how-to/upgrade-version.mdx

If your database engine is outdated, you can upgrade the version to the latest one anytime via the Scaleway console.

<Message type="note">

>   You must upgrade your engine version if your Database Instance uses an End of Life engine version that is no longer maintained.

Reviewer: Add a link here to the upgrade procedure?

- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- A [PostgreSQL Database Instance](/managed-databases-for-postgresql-and-mysql/how-to/create-a-database/) running on an outdated engine version

## How to remove incompatible data types

> Before upgrading to a major PostgreSQL version, you must remove any data types incompatible with `pg_upgrade`.

Reviewer 2: ```suggestion
Before upgrading to a newer major PostgreSQL version, you must remove any data types incompatible with `pg_upgrade`.
```

> Before upgrading to a newer major PostgreSQL version, you must remove any data types incompatible with `pg_upgrade`.

Reviewer 3: ```suggestion
Before upgrading to a newer major PostgreSQL version, you must remove any data types incompatible with `pg_upgrade` tool.
```

The `reg*` data type, for example, cannot be persisted by `pg_upgrade`.

> Run the following command in your Database Instance to remove all instances of the `reg*` data type:

Reviewer 3: ```suggestion
Run the following command in your Database Instance to identify all instances of the `reg*` data type and look to remove them:
```

    - **Upgrade only**: This method has no impact on your original Database Instance. Your Database Instance will remain available with its original endpoint and will continue to be billed.
    - **Upgrade and switch incoming traffic**: With this method, your endpoint is migrated to the new Database Instance automatically. This option will create a clone of your Database Instance, and will automatically migrate the endpoint to the new Instance. The original Database Instance remains available and will continue to be billed. The endpoint will be deleted from the original Database Instance.

    <Message type="important">

>       When you upgrade to a new version:

Reviewer 3: May be add an advice about the engine upgrade: customers can try a migration in a "dry-run" to test how many time this migration can take and check if the upgrade succeed. We don't touch about the source instance so they can test a major migration without impact if they don't use the migration endpoint checkbox/flag. And after this test, they can run the real migration with endpoint migration for example.

        - There is no synchronization between the source and target Database Instances. To avoid data loss, we recommend you stop any write operations running on your applications during the upgrade. If you let them run during the process, the data will be stored only in the source Database Instance.

>         - Your Database Instances in [High Availability (HA)](/managed-databases-for-postgresql-and-mysql/concepts/#high-availability) mode will migrate to a standalone Instance. To maintain your HA mode, you must manually [enable HA](/managed-databases-for-postgresql-and-mysql/how-to/enable-high-availability) after the upgrade is complete.

Reviewer 3: Same for read replica, there will not exist on the new instance.

>         - Database Instance [advanced settings](/managed-databases-for-postgresql-and-mysql/how-to/configure-advanced-settings) are synced as far as they are still available on the new database engine version.

Reviewer 2: ```suggestion
        - Database Instance [advanced settings](/managed-databases-for-postgresql-and-mysql/how-to/configure-advanced-settings) are synced as long as they are still available on the new database engine version.
```

    </Message>

    <Message type="tip">
      Follow the [migrating endpoints via the CLI](/managed-databases-for-postgresql-and-mysql/api-cli/migrating-endpoints/) procedure to quickly migrate your endpoints. Before doing so, we recommend running a dry-run of the Database Instance upgrade to validate the process.<br /><br />
      You can use the same CLI command to revert the endpoint migration anytime. <br /><br />
      Keep in mind that reverting the endpoint will not affect the data stored on the databases. This means that if some entries were added to the upgraded database, they will not be added back to the old version when you change the endpoints.
    </Message>
4. Click **Upgrade version**. A new Database Instance is created.

    <Message type="important">

>       - In PostgreSQL major upgrades the `pgaudit` and `pg_stat_statements` [engine extensions](/managed-databases-for-postgresql-and-mysql/reference-content/postgresql-extensions) will be removed. You must reinstall these extensions once the upgrade is complete.

Reviewer 2: ```suggestion
      - In PostgreSQL major upgrades, the `pgaudit` and `pg_stat_statements` [engine extensions](/managed-databases-for-postgresql-and-mysql/reference-content/postgresql-extensions) will be removed. You must reinstall these extensions once the upgrade is complete.
```

>       - In PostgreSQL major upgrades, the `pgaudit` and `pg_stat_statements` [engine extensions](/managed-databases-for-postgresql-and-mysql/reference-content/postgresql-extensions) will be removed. You must reinstall these extensions once the upgrade is complete.

Reviewer 3: ```suggestion
      - In PostgreSQL major upgrades, the `pgaudit` and `pg_stat_statements` [engine extensions](/managed-databases-for-postgresql-and-mysql/reference-content/postgresql-extensions) will be removed. You must reinstall these extensions once the upgrade is complete on each databases there are installed.
```

>       - If you are upgrading to PG16, we recommend you [reset your user passwords](/managed-databases-for-postgresql-and-mysql/how-to/add-users) from the console or using `psql`, [PostrgreSQL's CLI](https://docs.postgresql.fr/13/app-psql.html). You must create a new password that follows the **SCRAM-SHA-256** format. The previous versions' password format is no longer supported in PG16.

Reviewer 3: ```suggestion
      - If you are upgrading to PG16, we recommend you [reset your user passwords](/managed-databases-for-postgresql-and-mysql/how-to/add-users) from the console or using `psql`, [PostrgreSQL's CLI](https://docs.postgresql.fr/13/app-psql.html). You must create a new password that follows the **SCRAM-SHA-256** format. The previous versions' password format will not be supported in feature PostgreSQL major version.
```

Reviewer 3: may be use this link to the psql command https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-PASSWORD

Reviewer 3: > You must create a new password that follows the **SCRAM-SHA-256** format.

Not really, new passwords will be stored in the **SCRAM-SHA-256** format by default in PG16. Customers just need to reapply their password (via CLI or the console) and it will be stored in the correct format automatically.

>       - If you are upgrading to PG16, we recommend you to [reset your user passwords](/managed-databases-for-postgresql-and-mysql/how-to/add-users) from the console or using `psql`, [PostrgreSQL's CLI](https://docs.postgresql.fr/13/app-psql.html). You must create a new password that follows the **SCRAM-SHA-256** format. The previous versions' password format is no longer supported in PG16.

Reviewer 2: ```suggestion
      - If you are upgrading to PG16, we recommend you [reset your user passwords](/managed-databases-for-postgresql-and-mysql/how-to/add-users) from the console or using `psql`, [PostrgreSQL's CLI](https://docs.postgresql.fr/13/app-psql.html). You must create a new password that follows the **SCRAM-SHA-256** format. The previous versions' password format is no longer supported in PG16.
```
